docs/apis/streaming/windows.md

```diff
@@ -24,1023 +24,608 @@ specific language governing permissions and limitations
+window operation to take the elements that are currently in the window and pass them along to
+the window function to produce output for the firing window.
+
+Each `WindowAssigner` (except `GlobalWindows`) comes with a default trigger that should be
+appropriate for most use cases. For example, `TumblingEventTimeWindows` has an `EventTimeTrigger` as
+default trigger. This trigger simply fires once the watermark passes the end of a window.
+
+You can specify the trigger to be used by calling `trigger()` with a given `Trigger`. The
+whole specification of the windowed transformation would then look like this:
 
 <div class="codetabs" markdown="1">
 <div data-lang="java" markdown="1">
 {% highlight java %}
-nonKeyedStream
-    .windowAll(SlidingEventTimeWindows.of(Time.seconds(5), Time.seconds(1))
-    .trigger(CountTrigger.of(100))
-    .evictor(CountEvictor.of(10));
+DataStream<T> input = ...;
+
+input
+    .keyBy(<key selector>)
+    .window(<window assigner>)
+    .trigger(<trigger>)
+    .<windowed transformation>(<window function>);
 {% endhighlight %}
 </div>
 
 <div data-lang="scala" markdown="1">
 {% highlight scala %}
-nonKeyedStream
-    .windowAll(SlidingEventTimeWindows.of(Time.seconds(5), Time.seconds(1))
-    .trigger(CountTrigger.of(100))
-    .evictor(CountEvictor.of(10))
+val input: DataStream[T] = ...
+
+input
+    .keyBy(<key selector>)
+    .window(<window assigner>)
+    .trigger(<trigger>)
+    .<windowed transformation>(<window function>)
 {% endhighlight %}
 </div>
 </div>
 
-Basic window definitions are also available for windows on non-keyed streams:
+Flink comes with a few triggers out-of-box: there is the already mentioned `EventTimeTrigger` that
+fires based on the progress of event-time as measured by the watermark, The `ProcessingTimeTrigger`
+does the same but based on processing time and the `CountTrigger` fires once the number of elements
+in a window exceeds the given limit.
+
+<span class="label label-danger">Attention</span> By specifying a trigger using `trigger()` you
+are overwriting the default trigger of a `WindowAssigner`. For example, if you specify a
+`CountTrigger` for `TumblingEventTimeWindows` you will no longer get window firings based on the
+progress of time but only by count. Right now, you have to write your own custom trigger if
+you want to react based on both time and count.
+
+The internal `Trigger` API is still considered experimental but you can check out the code
+if you want to write your own custom trigger:
+{% gh_link /flink-streaming-java/src/main/java/org/apache/flink/streaming/api/windowing/triggers/Trigger.java "Trigger.java" %}.
+
+## Non-keyed Windowing
+
+You can also leave out the `keyBy()` when specifying a windowed transformation. This means, however,
+that Flink cannot process windows for different keys in parallel, essentially turning the
+transformation into a non-parallel operation.
+
+
+The basic structure of a non-keyed windowed transformation is as follows:
```

Comment:
How about this:

`<span class="label label-danger">Warning</span>` As mentioned in the introduction, non-keyed
windows have the disadvantage that work cannot be distributed in the cluster because
windows cannot be computed independently per key. This can have severe performance implications.